### PR TITLE
feat: metrics hashrate

### PR DESCRIPTION
**Pull request by hunjixin**
### Review Type Requested (choose one):
- [x] **Glance** - superficial check (from domain experts)
- [ ] **Logic** - thorough check (from everybody doing review)

### Summary
Provide a one line summary and link to any relevant references

### Task/Issue reference

Closes: add_link_here

### Details (optional)
Add any additional details that might help Code Reviewers digest this PR

### CODE REVIEW

**Comment by hunjixin** [comment]
close https://github.com/Lilypad-Tech/internal/issues/133

**Review by richardbremner — APPROVED** [review]
lgtm

File: pkg/metricsDashboard/logger.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 
 const jobsEndpoint = "jobs"
 const nodeInfoEndpoint = "nodes"
+const hashrateEndpoint = "pow-logs/hashrates"
```
Comment (richardbremner) [inline pkg/metricsDashboard/logger.go]: this doesn't seem to match the new route in the API, can you please double check @hunjixin

File: pkg/resourceprovider/resourceprovider.go [hunk 2]
```diff
@@ -128,7 +132,30 @@ func (resourceProvider *ResourceProvider) StartMineLoop(ctx context.Context) cha
 		}
 	})
 
-	submitWork := func(nonce *big.Int) {
+	submitWork := func(nonce *big.Int, hashrate float64) {
```
Comment (richardbremner) [inline pkg/resourceprovider/resourceprovider.go:134]: it's not obvious whether or not this function works correctly, could you extract this to a unit test?

File: pkg/resourceprovider/minerctl.go [hunk 3]
```diff
@@ -201,11 +204,15 @@ out:
 				log.Warn().Msg("This work has been submit before")
 				continue
 			}
-			m.submit(result.Nonce.ToBig())
+
+			hashrate := float64(m.totalHash/1000/1000) / (float64(time.Since(workStartTime).Milliseconds()) / 1000.0)
+			m.submit(result.Nonce.ToBig(), hashrate)
 			stopWrokers()
```
Comment (AquiGorka) [inline pkg/resourceprovider/minerctl.go]: [nit] I just noticed there is a typo here: `stopWrokers` -> `stopWorkers`

File: pkg/resourceprovider/resourceprovider.go [hunk 4]
```diff
@@ -128,7 +132,30 @@ func (resourceProvider *ResourceProvider) StartMineLoop(ctx context.Context) cha
...
+
+		buf := make([]byte, 8)
+		_ = binary.PutVarint(buf, finishTime)
+		_, err = hasher.Write(buf)
+		if err != nil {
+			log.Err(err).Msgf("Write finish time")
+			return
+		}
+
+		id := hex.EncodeToString(hasher.Sum(nil))
```
Comment (AquiGorka) [inline pkg/resourceprovider/resourceprovider.go]: Would it make more sense to use a similar approach to the other ids here?
Setting the id to `address` + `date`? This way we can prevent multiple entries for the same miner on the same date no?

File: pkg/metricsDashboard/logger.go [hunk 5]
```diff
@@ -103,3 +104,14 @@ func TrackDeal(params DealPayload) {
 	url := host + namespace + "/" + dealsEndpoint
 	http.GenericJSONPostClient(url, payload)
 }
+
+func TrackHashrate(hashrate data.MinerHashRate) {
```
Comment (AquiGorka) [inline pkg/metricsDashboard/logger.go]: Please move this function to the [`powLogs` package](https://github.com/Lilypad-Tech/lilypad/blob/main/pkg/powLogs/logger.go)

File: pkg/resourceprovider/resourceprovider.go (hunk 4 above)
Comment (hunjixin) [reply pkg/resourceprovider/resourceprovider.go]: change to ```walletAddress.String() + strconv.FormatInt(finishTime, 10)```.  but why not use combined primary key

File: pkg/metricsDashboard/logger.go (hunk 1 above)
Comment (richardbremner) [reply pkg/metricsDashboard/logger.go]: I checked the api repo, it looks ok

**Comment by hunjixin** [comment]
image

**Review by richardbremner — APPROVED** [review]
lgtm

**Pull request merged**